apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -552,20 +584,43 @@ private void AssertOpen(string message)
                 throw new InvalidOperationException("The connection must be open.  Failed to " + message);
         }
 
-        private Task<SpannerTransaction> BeginTransactionImpl(CancellationToken cancellationToken,
-            TransactionOptions transactionOptions,
-            TransactionMode transactionMode)
+        internal Task<SingleUseTransaction> BeginSingleUseTransactionAsync(
+            TimestampBound targetReadTimeStamp,
+            CancellationToken cancellationToken)
+        {
+            var options = ConvertToOptions(targetReadTimeStamp);
+            return ExecuteHelper.WithErrorTranslationAndProfiling(async () =>
+            {
+                using (var sessionHolder = await SessionHolder.Allocate(this,
+                        options, cancellationToken)
+                    .ConfigureAwait(false))
+                {
+                    await sessionHolder.Session.RemoveFromTransactionPoolAsync();
```

Comment:
ConfigureAwait :)
(We could rebase this to master, at which point you'll start getting errors, as I've added a checker.)